packages/rich-text-from-markdown/src/index.ts

```diff
@@ -84,64 +85,79 @@ const isInline = (nodeType: string) => {
   return nodeContainerTypes.get(nodeType) === 'inline';
 };
 
-async function mdToRichTextNode(node: MarkdownNode, fallback: FallbackResolver): Promise<Node> {
+async function mdToRichTextNode(
+  node: MarkdownNode,
+  fallback: FallbackResolver,
+  appliedMarksTypes: string[] = [],
+): Promise<Node[]> {
   const nodeType = nodeTypeFor(node);
 
   if (isLink(node)) {
-    const content = (await mdToRichTextNodes(node.children, fallback)) as Text[];
+    const content = (await mdToRichTextNodes(node.children, fallback, appliedMarksTypes)) as Text[];
 
     const hyperlink: Hyperlink = {
       nodeType: INLINES.HYPERLINK,
       data: { uri: node.url },
       content,
     };
 
-    return hyperlink;
+    return [hyperlink];
   }
   if (isBlock(nodeType) || isInline(nodeType)) {
-    const content = await mdToRichTextNodes(node.children, fallback);
+    const content = await mdToRichTextNodes(node.children, fallback, appliedMarksTypes);
 
-    return {
-      nodeType: nodeType,
-      content,
-      data: {},
-    } as Block | Inline;
+    return [
+      {
+        nodeType: nodeType,
+        content,
+        data: {},
+      } as Block | Inline,
+    ];
   } else if (isText(nodeType)) {
-    let marks = [];
     let nodeValue = node.value;
+    const markType = markTypeFor(node);
+    const marks = [...appliedMarksTypes];
+    if (markType) {
+      marks.push(markType);
+    }
     if (node.type !== 'text') {
-      // TODO: this implementation needs to handle arbitrarily nested marks
-      // for example: **_Hello_, world!**
-      // this a markdown node here that's why we use children
-      nodeValue = node.children ? node.children[0].value : node.value;
-
-      const markType = markTypeFor(node);
-      if (markType) {
-        marks.push({
-          type: markTypeFor(node),
-        });
+      if (node.children) {
+        return await mdToRichTextNodes(node.children, fallback, marks);
       }
     }
-    return {
-      nodeType: nodeType,
-      value: nodeValue,
-      marks: marks,
-      data: {},
-    } as Text;
+
+    if (node.value) {
+      return [
+        {
```

Comment:
I think it would be better if transform these 

```
return [
  {
	.....
  }
]
```

to a factory function like 

```js
return buildNode('text', options)
```
WDYT

Reply:
yes, sounds good